src/qt/listctrl.cpp

```diff
@@ -141,96 +144,868 @@ class wxQtTreeWidget : public wxQtEventSignalHandler< QTreeWidget, wxListCtrl >
+            std::advance(i, row);
+            m_rows.insert(i, RowItem(m_headers.size()));
+            newRowIndex = row;
+        }
+
+        beginInsertRows(QModelIndex(), newRowIndex, newRowIndex);
+
+        info.SetId(newRowIndex);
+        SetItem(info);
+
+        endInsertRows();
+
+        return newRowIndex;
+    }
+
+    long InsertColumn(long col, const wxListItem& info)
+    {
+        long newColumnIndex;
+
+        ColumnItem newColumn;
+        newColumn.m_align = wxQtConvertTextAlign(info.GetAlign());
+        newColumn.m_label = wxQtConvertString(info.GetText());
+
+        if ( col == -1 || static_cast<size_t>(col) >= m_headers.size() )
+        {
+            newColumnIndex = m_headers.empty() ? 0 : m_headers.size() - 1;
+        }
+        else
+        {
+            newColumnIndex = col;
+        }
+
+        beginInsertColumns(QModelIndex(), newColumnIndex, newColumnIndex);
+
+        std::vector<ColumnItem>::iterator i = m_headers.begin();
+        std::advance(i, newColumnIndex);
+        m_headers.insert(i, newColumn);
+
+        const int numberOfRows = m_rows.size();
+
+        for (int i = 0; i < numberOfRows; ++i )
+        {
+            std::vector<ColumnItem>::iterator it = m_rows[i].m_columns.begin();
+            std::advance(it, newColumnIndex);
+            m_rows[i].m_columns.insert(it, newColumn);
+        }
+
+        endInsertColumns();
+
+        return true;
+    }
+
+    void SortItems(wxListCtrlCompare fn, wxIntPtr data)
+    {
+        CompareAdapter compare(fn, data);
+        std::sort(m_rows.begin(), m_rows.end(), compare);
+    }
+
+    bool IsItemChecked(long item) const
+    {
+        wxCHECK_MSG(item >= 0 && static_cast<size_t>(item) <= m_rows.size(),
+            false, "Invalid row");
+
+        return m_rows[item].m_checked;
+    }
+
+    void CheckItem(long item, bool check)
+    {
+        wxCHECK_RET(item >= 0 && static_cast<size_t>(item) <= m_rows.size(),
+            "Invalid row");
+
+        m_rows[item].m_checked = check;
+
+        QVector<int> roles;
+        roles.push_back(Qt::CheckStateRole); 
+        const QModelIndex modelIndex = index(item,0);
+        dataChanged(modelIndex, modelIndex, roles);
+    }
+
+    virtual bool IsVirtual() const
+    {
+        return false;
+    }
+
+    virtual void SetVirtualItemCount(long WXUNUSED(count))
+    {
+    }
+
+protected:
+    wxImageList *GetImageList() const
+    {
+        const int requiredList = m_listCtrl->HasFlag(wxLC_SMALL_ICON)
+            ? wxIMAGE_LIST_SMALL
+            : wxIMAGE_LIST_NORMAL;
+        return m_listCtrl->GetImageList(requiredList);
+    }
+
+    wxListCtrl *GetListCtrl() const
+    {
+        return m_listCtrl;
+    }
+
+    bool IsSelected(const QModelIndex& index) const
+    {
+        QModelIndexList indices = m_view->selectionModel()->selectedIndexes();
+        return indices.contains(index);
+    }
+
+    void CopySelectStatusToItem(wxListItem& info, const int row, const int col)
```

Comment:
Top level const is useless and ignored anyhow:
```suggestion
    void CopySelectStatusToItem(wxListItem& info, int row, int col)
```